### PR TITLE
feat(cdk/overlay): connected-overlay directive should have input for disabling escape close #19420

### LINKED ISSUE #19420: feat(cdk/overlay): connected-overlay directive should have input for disabling escape close
#### Feature Description

Currently connected-overlay is hardcoded to be closed on escape. https://github.com/angular/components/blob/5f125395e0d39eb7e6a34d0c3355fc474f4d37a8/src/cdk/overlay/overlay-directives.ts#L287-L289
I propose adding a way to control this behavior.

#### Use Case

In my project I need an overlay that does not close on keyboard event. Also, if in the overlay we have a form, there may happen a situation when user should be forced to complete it before closing the overlay.

I create overlay in this manner:
```
  cdk-connected-overlay
  cdkConnectedOverlayHasBackdrop
  cdkConnectedOverlayBackdropClass="light-backdrop"
  [cdkConnectedOverlayOrigin]="origin"
  [cdkConnectedOverlayPositions]="positions"
  [cdkConnectedOverlayOpen]="isOpen"
  [cdkConnectedOverlayScrollStrategy]="scrollStrategy"
```

### PR DESCRIPTION
**Pull request by victorward**
#19420

### CODE REVIEW

**Comment by victorward** [comment]
#19420

File: src/cdk/overlay/overlay-directives.ts [hunk 1]
```diff
@@ -181,6 +181,9 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
   /** Whether the overlay is open. */
   @Input('cdkConnectedOverlayOpen') open: boolean = false;
 
+  /** Whether the overlay is open. */
+  @Input('cdkConnectedOverlayCloseOnEscape') closeOnEscape: boolean = true;
```
Comment (crisbeto) [inline src/cdk/overlay/overlay-directives.ts:185]: We have something similar in the dialog, bottom sheet and sidenav called `disableClose` which we should probably align with. Note that in the other components it means disabling close when pressing escape and clicking on the backdrop.

File: src/cdk/overlay/overlay-directives.ts [hunk 2]
```diff
@@ -181,6 +181,9 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
   /** Whether the overlay is open. */
   @Input('cdkConnectedOverlayOpen') open: boolean = false;
 
+  /** Whether the overlay is open. */
```
Comment (crisbeto) [inline src/cdk/overlay/overlay-directives.ts:184]: The doc is incorrect.

File: src/cdk/overlay/overlay-directives.ts (hunk 1 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay-directives.ts:185]: +1, we should be consistent with the name and the behavior

Comment (victorward) [reply src/cdk/overlay/overlay-directives.ts:185]: Didn't know about `disableClose`, totally agree with it, I will modify PR soon

File: src/cdk/overlay/overlay-directives.ts (hunk 2 above)
Comment (victorward) [reply src/cdk/overlay/overlay-directives.ts]: Fixed

File: src/cdk/overlay/overlay-directives.ts (hunk 1 above)
Comment (victorward) [reply src/cdk/overlay/overlay-directives.ts]: @crisbeto I refactored code, also align behavior as it is in Dialog.
But `overlay-directive` doesn't have close action on backdrop [click](https://github.com/angular/components/blob/f21acde8341ecff7731aeac044f8ced3b1e08459/src/cdk/overlay/overlay-directives.ts#L394), I verified it manually also created [test](https://github.com/angular/components/pull/20585/commits/ab9d94fd4e82e060d9e180123e56c3e89dffb298#diff-2988ed2cf46548b0ded8e4be73b86babR178), that shows current behavior

**Comment by victorward** [comment]
@jelbourn @crisbeto Can you, please, review it again?
(All comments resolved)

**Review by jelbourn — APPROVED** [review]
LGTM

**Pull request merged**